app/models/efile_submission.rb

```diff
@@ -141,6 +141,14 @@ def previously_transmitted_submission
     previous_submission if previous_submission.last_transition_to("transmitted").present?
   end
 
+  def source_record
+    if ["StateFileNyIntake", "StateFileAzIntake"].include? data_source_type
```

Comment:
we could potentially improve this line by saying
```ruby
is_for_state_filing? ? data_source : tax_return
```

Reply:
I hadn't realized that there were already methods to tell if it was for state vs federal filing

Reply:
but this is definitely not blocking!

Reply:
this is a good change! added it